firefox-ios/Client/Frontend/StoriesFeed/StoriesWebviewViewController.swift

```diff
@@ -105,6 +128,38 @@ class StoriesWebviewViewController: UIViewController,
         navigationController?.navigationBar.compactAppearance = appearance
     }
 
+    // MARK: - WKNavigationDelegate
+    @MainActor
+    func webView(
+        _ webView: WKWebView,
+        decidePolicyFor navigationAction: WKNavigationAction,
+        decisionHandler: @escaping @MainActor (WKNavigationActionPolicy) -> Void
+    ) {
+        decisionHandler(.allow)
+        return
+    }
+
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation?) {
+        // Update domain label when navigation finishes (in the same window)
+        domainLabel.text = webView.url?.normalizedHost
+    }
+
+    // MARK: - WKUIDelegate
+    func webView(
+        _ webView: WKWebView,
+        createWebViewWith configuration: WKWebViewConfiguration,
+        for navigationAction: WKNavigationAction,
+        windowFeatures: WKWindowFeatures
+    ) -> WKWebView? {
+        // Update domain label when opening a link that would normally open in another webview
+        domainLabel.text = navigationAction.request.url?.normalizedHost
+
+        // If the page uses `window.open()` or `[target="_blank"]`, continue to open the page in the current webview
+        // since there is no concept of tab management in the stories experience
+        webView.load(navigationAction.request)
```

Comment:
I wonder if we could also test this with `MockWKWebView`. Not urgent, just to keep in mind as I don't know what features we want to support in this stories mode.